private/buf/bufprint/bufprint.go

```diff
@@ -73,66 +78,182 @@ func (f Format) String() string {
-// NewCuratedPluginPrinter returns a new CuratedPluginPrinter.
-func NewCuratedPluginPrinter(writer io.Writer) CuratedPluginPrinter {
-	return newCuratedPluginPrinter(writer)
+// PrintNames prints entities' names.
+//
+// If format is FormatJSON, this also prints information about each entity, the
+// same as calling PrintInfo on each entity.
+func PrintNames(writer io.Writer, format Format, entities ...OutputEntity) error {
+	switch format {
+	case FormatJSON:
+		for _, entity := range entities {
+			if err := json.NewEncoder(writer).Encode(entity); err != nil {
+				return err
+			}
+		}
+		return nil
+	case FormatText:
+		for _, entity := range entities {
+			if _, err := fmt.Fprintln(writer, entity.fullName()); err != nil {
+				return err
+			}
+		}
+		return nil
+	default:
+		return syserror.Newf("unknown format: %s", format)
+	}
 }
 
-// OrganizationPrinter is an organization printer.
-type OrganizationPrinter interface {
-	PrintOrganizationInfo(ctx context.Context, format Format, organization *ownerv1.Organization) error
+// PrintPage prints a page of entities.
+func PrintPage(
+	writer io.Writer,
+	format Format,
+	nextPageToken string,
+	nextPageCommand string,
+	entities []OutputEntity,
+) error {
+	if len(entities) == 0 {
+		return nil
+	}
+	var entitiesName string
+	for _, entity := range entities {
+		var currentEntitiesName string
+		switch entity.(type) {
+		case outputLabel:
+			currentEntitiesName = "labels"
+		case outputCommit:
+			currentEntitiesName = "commits"
+		case outputModule:
+			currentEntitiesName = "modules"
+		case outputOrganization:
+			currentEntitiesName = "organizations"
+		default:
+			return syserror.Newf("unknown implementation of OutputEntity: %T", entity)
+		}
+		if currentEntitiesName != entitiesName && entitiesName != "" {
+			return syserror.Newf("the page has both %s and %s", currentEntitiesName, entitiesName)
+		}
+		entitiesName = currentEntitiesName
+	}
+	switch format {
+	case FormatText:
+		if err := PrintNames(writer, format, entities...); err != nil {
+			return err
+		}
+		if nextPageToken == "" {
+			return nil
+		}
+		_, err := fmt.Fprintf(
+			writer,
+			"\nMore than %d %s found, run %q to list more\n",
+			len(entities),
+			entitiesName,
+			nextPageCommand,
+		)
+		return err
+	case FormatJSON:
+		return json.NewEncoder(writer).Encode(&entityPage{
+			NextPage:         nextPageToken,
+			Entities:         entities,
+			pluralEntityName: entitiesName,
+		})
+	default:
+		return syserror.Newf("unknown format: %v", format)
+	}
 }
 
-// NewOrganizationPrinter returns a new OrganizationPrinter.
-func NewOrganizationPrinter(address string, writer io.Writer) OrganizationPrinter {
-	return newOrganizationPrinter(address, writer)
+// Print prints an entity's information.
+//
+// If format is FormatText, this prints the information in a table.
+// If format is FormatJSON, this prints the information as a JSON object.
+func PrintInfo(writer io.Writer, format Format, entity OutputEntity) error {
```

Comment:
PrintEntity

Reply:
fixed in [1aac338](https://github.com/bufbuild/buf/pull/3173/commits/1aac338032f18da66e0da1540f399a51092a5a0f)